src/System.Drawing.Common/src/System/Drawing/Font.cs

```diff
@@ -275,8 +275,70 @@ public override string ToString()
                                     _gdiCharSet,
                                     _gdiVerticalFont);
         }
-        
+
         // This is used by SystemFonts when constructing a system Font objects.
         internal void SetSystemFontName(string systemFontName) => _systemFontName = systemFontName;
+
+        public unsafe void ToLogFont(object logFont, Graphics graphics)
+        {
+            if (logFont == null)
+            {
+                throw new ArgumentNullException(nameof(logFont));
+            }
+
+            Type type = logFont.GetType();
+            int nativeSize = sizeof(SafeNativeMethods.LOGFONT);
+            if (Marshal.SizeOf(type) != nativeSize)
+            {
+                // If we don't actually have an object that is LOGFONT in size, trying to pass
+                // it to GDI+ is likely to cause an AV.
+                throw new ArgumentException();
+            }
+
+            SafeNativeMethods.LOGFONT nativeLogFont = ToLogFontInternal(graphics);
+
+            if (!type.IsValueType)
```

Comment:
@jkotas I changed this so round tripping blittable value types works. Is there a better way to do this?

Reply:
So either type below works now:

``` C#
[StructLayout(LayoutKind.Sequential, CharSet = CharSet.Unicode)]
public unsafe struct LOGFONT
{
    private const int LF_FACESIZE = 32;

    public int lfHeight;
    public int lfWidth;
    public int lfEscapement;
    public int lfOrientation;
    public int lfWeight;
    public byte lfItalic;
    public byte lfUnderline;
    public byte lfStrikeOut;
    public byte lfCharSet;
    public byte lfOutPrecision;
    public byte lfClipPrecision;
    public byte lfQuality;
    public byte lfPitchAndFamily;
    private fixed char _lfFaceName[LF_FACESIZE];
}

[StructLayout(LayoutKind.Sequential, CharSet = CharSet.Unicode)]
public unsafe class LOGFONTCLASS
{
    private const int LF_FACESIZE = 32;

    public int lfHeight;
    public int lfWidth;
    public int lfEscapement;
    public int lfOrientation;
    public int lfWeight;
    public byte lfItalic;
    public byte lfUnderline;
    public byte lfStrikeOut;
    public byte lfCharSet;
    public byte lfOutPrecision;
    public byte lfClipPrecision;
    public byte lfQuality;
    public byte lfPitchAndFamily;
    [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 32)]
    public string lfFaceName;
}
```

Reply:
The only way to maintain full compatibility with what used to be there is to keep the `AsAny` marshaling. I think anything else it going to be broken for some set of cases that used to work before.

Reply:
> So either type below works now:

Does this work?

```
[StructLayout(LayoutKind.Sequential, CharSet = CharSet.Unicode)]
public struct LOGFONT
{
    private const int LF_FACESIZE = 32;

    public int lfHeight;
    public int lfWidth;
    public int lfEscapement;
    public int lfOrientation;
    public int lfWeight;
    public byte lfItalic;
    public byte lfUnderline;
    public byte lfStrikeOut;
    public byte lfCharSet;
    public byte lfOutPrecision;
    public byte lfClipPrecision;
    public byte lfQuality;
    public byte lfPitchAndFamily;
    [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 32)]
    public string lfFaceName;
}
```

Reply:
If it is important for AsAny marshaling on bllitable types to be faster, the runtime is best place to fix it. I think you trying to special case here is losing battle unless you are willing to resort to breaking changes.

Having said this, I doubt that the AsAny marshaling overhead would show up here.

Reply:
Ok, I'm going for breaking here (unblittable value types won't work). I finished changing the `LOGFONT` references and there is just too much benefit to keeping it blittable internally. We can add a P/Invoke overload in the future if we really need to but I'd rather not put that in unless we really need it.

Finishing tidying up the `LOGFONT` references I see a significant improvement in getting `SystemFonts.DefaultFont` (which is in WinForms startup). Just getting that property goes from 33ms/3920bytes down to 22ms/2000bytes.

`SystemFonts.MenuFont` (which WinForms uses) goes from 38ms/4192bytes down to 24ms/1936bytes.

Reply:
What is the impact of this breaking change? Performance improvement alone is not sufficient for doing a breaking change.

Reply:
> What is the impact of this breaking change? 

If, for some reason, you created your own data structure for this that was a non-blittable struct you would no longer work. We never documented the internals of the object returned- if you grab and return it through the APIs then you will still work. If you grab an old versions and somehow get it to a new version, it will work- and vice-versa.

Reply:
Agree that this looks like benign breaking. I assume that you are going to apply similar cleanup to the WinForms side as well - so that the AsAny marshalling is eliminated there too.